scripts/process_readme.py

```diff
@@ -3,126 +3,100 @@
-            new_str = "{0}/{1}".format(get_base_path(target_version), m)
+            new_str = "{0}/{1}".format(_get_base_path(target_version), m)
             result = result.replace(old_str, new_str)
-        logger.info("Updated string: %s", result.rstrip())
+        _logger.info("Updated string: %s", result.rstrip())
 
     return result
 
 
-def update_same_page_references(line, target_version):
-    logger.debug("Starting %s", sys._getframe().f_code.co_name)
+def _update_same_page_references(line, target_version):
+    _logger.info("Starting %s", sys._getframe().f_code.co_name)
     same_page_ref_pattern = re.compile(_SAME_MD_REF_PATTERN)
     result = line
     match = same_page_ref_pattern.search(line)
     if match:
-        logger.info("Matched %s", match)
+        _logger.info("Matched %s", match)
         for m in match.groups():
             old_str = m
-            new_str = "{0}{1}".format(get_base_path(target_version), m)
+            new_str = "{0}{1}".format(_get_base_path(target_version), m)
             result = result.replace(old_str, new_str)
-        logger.info("Updated string: %s", result.rstrip())
+        _logger.info("Updated string: %s", result.rstrip())
 
     return result
 
 
-def process_line(line, target_version):
-    logger.debug("Starting %s", sys._getframe().f_code.co_name)
-    result = update_current_version(line, target_version)
-    result = update_other_markdown_references(result, target_version)
-    result = update_same_page_references(result, target_version)
+def _process_line(line, target_version):
+    _logger.info("Starting %s", sys._getframe().f_code.co_name)
+    result = _update_current_version(line, target_version)
+    result = _update_other_markdown_references(result, target_version)
+    result = _update_same_page_references(result, target_version)
 
     return result
 
 
-def main(argv):
-    parser = build_argument_parser()
-    args = parser.parse_args(argv)
-
-    if args.loglevel:
-        logging.basicConfig(level=getattr(logging, args.loglevel))
-
-    target_version = get_fairlearn_version()
-    logger.info("fairlearn version: %s", target_version)
+def process_readme(input_file_name, output_file_name):
+    import fairlearn
+    target_version = fairlearn.__version__
+    _logger.info("fairlearn version: %s", target_version)
 
-    logger.debug("Reading file %s", args.input)
     text_lines = []
-    with open(args.input, 'r') as f:
-        text_lines = f.readlines()
-
-    result_lines = [process_line(l, target_version) for l in text_lines]
+    with _LogWrapper("reading file {}".format(input_file_name)):
 
-    logger.debug("Writing file %s", args.output)
-    with open(args.output, 'w') as f:
-        f.writelines(result_lines)
-    logger.debug("Completed")
+        with open(input_file_name, 'r') as input_file:
+            text_lines = input_file.readlines()
 
+    result_lines = [_process_line(line, target_version) for line in text_lines]
 
-if __name__ == "__main__":
```

Comment:
No longer a standalone script?

Reply:
It could be used as such actually, let me fix that